packages/nimble-components/src/table-column/number-text/index.ts

```diff
@@ -0,0 +1,92 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { attr } from '@microsoft/fast-element';
+import { styles } from '../base/styles';
+import { template } from '../base/template';
+import type { TableNumberField } from '../../table/types';
+import { TableColumnTextBase } from '../text-base';
+import { TableColumnSortOperation } from '../base/types';
+import { tableColumnNumberTextGroupHeaderTag } from './group-header-view';
+import { tableColumnNumberTextCellViewTag } from './cell-view';
+import type { ColumnInternalsOptions } from '../base/models/column-internals';
+import { NumberTextFormat, defaultNumberOfDigits } from './types';
+
+export type TableColumnNumberTextCellRecord = TableNumberField<'value'>;
+export interface TableColumnNumberTextColumnConfig {
+    formatter: Intl.NumberFormat;
+    exponentialFormatter?: Intl.NumberFormat;
+}
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-number-text': TableColumnNumberText;
+    }
+}
+
+/**
+ * The table column for displaying numbers as text.
+ */
+export class TableColumnNumberText extends TableColumnTextBase {
+    @attr
+    public format: NumberTextFormat;
+
+    public override connectedCallback(): void {
+        super.connectedCallback();
+        this.updateColumnConfig();
+    }
+
+    protected override getColumnInternalsOptions(): ColumnInternalsOptions {
+        return {
+            cellRecordFieldNames: ['value'],
+            cellViewTag: tableColumnNumberTextCellViewTag,
+            groupHeaderViewTag: tableColumnNumberTextGroupHeaderTag,
+            delegatedEvents: [],
+            sortOperation: TableColumnSortOperation.basic
+        };
+    }
+
+    private formatChanged(): void {
+        this.updateColumnConfig();
+    }
+
+    private updateColumnConfig(): void {
+        this.columnInternals.columnConfig = this.createFormatterConfig();
```

Comment:
How would you feel about encapsulating the Formatter behaviors?

Something like:
```ts

class NumberFormatter {
    abstract format (number: number);
}

class DefaultFormatter extends NumberFormatter {
    /* encapsulates all the logic around doing the auto formatting*/
}

class IntegerFormatter extends NumberFormatter {
   /* same, encapsulates the integer formatting stuff */
}

class CustomFormatter extends NumberFormatter {
    /* just pass the formatter reference */
    constructor(private formatter: Intl.NumberFormat) { }
}

And then we just pass the specific instance via columnConfig. Don't need to spread out the state for specific types of formatters on the columnConfig
```